eox_nelp/user_authn/tests/test_views.py

```diff
@@ -0,0 +1,166 @@
+"""This file contains all the test for the Nelp registration form factory.
+Classes:
+    RegistrationFormFactoryTestCase: Test NelpRegistrationFormFactory login and class manipulation.
+"""
+from django.test import TestCase
+from django.utils.translation import gettext_lazy as _
+from mock import Mock
+
+from eox_nelp.edxapp_wrapper.site_configuration import configuration_helpers
+from eox_nelp.edxapp_wrapper.user_api import accounts
+from eox_nelp.user_authn.views.registration_form import NelpRegistrationFormFactory
+
+
+def mock_responses(responses):
+    """Function that manage the response of a mock based in its args.
+    You configre using a dict defining in the dict they key of each the args is used the response
+    of the function would be.
+    eg:
+    my_mock.foo.side_effect = mock_responses({
+        'x': 43,
+        'y': [4,5,6]
+    })
+    my_mock.foo('x') # => 43
+    my_mock.foo('y') # => [4,5,6]
+    my_mock.foo('diferent_arg') # => None
+    """
+    return lambda input, default_value=None: responses[input] if input in responses else default_value
+
+
+class RegistrationFormFactoryTestCase(TestCase):
+    """Test for Nelp registration form factory"""
+    EXTRA_FIELDS = [
+        "confirm_email",
+        "first_name",
+        "last_name",
+        "city",
+        "state",
+        "country",
+    ]
+
+    def setUp(self):
+        """
+        Create site since the view use the request.site attribute to determine the current domain.
```

Comment:
You are not creating any site

Reply:
[310561e](https://github.com/eduNEXT/eox-nelp/pull/138/commits/310561e072130041b4fa2ef887ed430225e1096a)